### PR TITLE
feat: transfer returned amount in DAIJoinMock

### LINKED ISSUE #41: Transfer returned amount in `DaiJoinMock`
Currently during unit and staging tests the excess payment amount stays in the contract.

### PR DESCRIPTION
**Pull request by ivaylonikolov7**
resolves #41

### CODE REVIEW

File: test/DssVestTopUp.test.ts [hunk 1]
```diff
@@ -156,11 +157,10 @@ describe("DssVestTopUp", function () {
       });
 
       it("should vest accrued tokens", async function () {
-        const unpaid = await dssVest.unpaid(vestId);
         await topUp.refundUpkeep();
         const vested = await token.balanceOf(topUp.address);
 
-        expect(unpaid).to.eq(vested);
+        expect(vested).to.eq(maxDepositAmt);
```
Comment (imollov) [inline test/DssVestTopUp.test.ts:163]: What's the rationale behind this change?

Comment (ivaylonikolov7) [reply test/DssVestTopUp.test.ts:163]: Before these changes, there were 10 000 tokens left in the contract but now that the surplus is transferred we are only left with maxDepositAmt in the topUp contract.

File: contracts/DssVestTopUp.sol [hunk 2]
```diff
@@ -137,6 +138,7 @@ contract DssVestTopUp is IUpkeepRefunder, Ownable {
             if (amt > maxDepositAmt) {
                 // Return excess amount to surplus buffer
                 uint256 excessAmt = amt - maxDepositAmt;
+                IERC20(paymentToken).approve(address(daiJoin), excessAmt);
```
Comment (imollov) [inline contracts/DssVestTopUp.sol:141]: Please use `TransferHelper.safeApprove` like in `_swapPaymentToLink` function.

File: contracts/DssVestTopUp.sol [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.9;
 
+import "@openzeppelin/contracts/token/ERC20/presets/ERC20PresetMinterPauser.sol";
```
Comment (imollov) [inline contracts/DssVestTopUp.sol:4]:
```suggestion
```

**Review by imollov — APPROVED** [review]

File: test/DssVestTopUp.test.ts [hunk 4]
```diff
@@ -177,7 +177,9 @@ describe("DssVestTopUp", function () {
           ["address", "uint256"],
           joinEvent?.data || ""
         );
+        const tokensInVowAddr = await token.balanceOf(vowAddr);
```
Comment (imollov) [inline test/DssVestTopUp.test.ts:180]:
```suggestion
        const surplusBufferBalance = await token.balanceOf(vowAddr);
```

File: test/DssVestTopUp.test.ts [hunk 5]
```diff
@@ -177,7 +177,9 @@ describe("DssVestTopUp", function () {
...
 
+        expect(tokensInVowAddr).to.equal(vestedTokens.sub(maxDepositAmt));
```
Comment (imollov) [inline test/DssVestTopUp.test.ts:182]: This is assuming all vested tokens are accrued at this point. 

Also, if you're changing the approach why is this assertion not changed to use `vestedTokens`?

`expect(joinAmt).to.eq(unpaid.sub(maxDepositAmt));`

Comment (ivaylonikolov7) [reply test/DssVestTopUp.test.ts:182]: Fair point but we are going to have some repeating `expect` functions. 

Therefore I think removing the event in the DaiJoinMock contract is going to be beneficial as we can do the same when we check the balance of the vow contract address.

File: contracts/test/DaiJoinMock.sol [hunk 6]
```diff
@@ -1,10 +1,24 @@
...
+        address from,
+        address to, 
+        uint256 amount
+    ) external;
+}
+
 contract DaiJoinMock {
+    ERC20Like paymentToken;
+    
     event Join(address usr, uint256 wad);
```
Comment (imollov) [inline contracts/test/DaiJoinMock.sol:16]:
```suggestion
```

File: test/DssVestTopUp.test.ts [hunk 7]
```diff
@@ -156,30 +157,16 @@ describe("DssVestTopUp", function () {
...
-        const [vowAddr, joinAmt] = abiCoder.decode(
-          ["address", "uint256"],
-          joinEvent?.data || ""
-        );
-
-        expect(vowAddr).to.eq(fakeVow);
-        expect(joinAmt).to.eq(unpaid.sub(maxDepositAmt));
+        await topUp.refundUpkeep();
+        const surplusBufferBalance = await token.balanceOf(fakeVow);
+        expect(surplusBufferBalance).to.equal(vestedTokens.sub(maxDepositAmt));
```
Comment (imollov) [inline test/DssVestTopUp.test.ts:169]: As stated in a previous comment, using `vestedTokens` is assuming the whole amount is accrued and that's why `unpaid` is called.

**Review by imollov — APPROVED** [review]
